pkg/controllers/user/pipeline/hooks/driver.go

```diff
@@ -0,0 +1,18 @@
+package hooks
+
+import (
+	"github.com/rancher/rancher/pkg/controllers/user/pipeline/hooks/drivers"
+	"github.com/rancher/types/config"
+	"net/http"
+)
+
+var Drivers map[string]Driver
+
+type Driver interface {
+	Execute(req *http.Request) (int, error)
+}
+
+func RegisterDrivers(Management *config.ManagementContext) {
+	Drivers = map[string]Driver{}
+	Drivers[drivers.GithubWebhookHeader] = drivers.GithubDriver{Management: Management}
```

Comment:
Lets avoid passing the entire management context. Pass API interfaces/listers that are gonna be used by drivers.